src/replica/replication_app_base.h

```diff
@@ -71,13 +72,14 @@ class replica_init_info
 
 public:
     replica_init_info() { memset((void *)this, 0, sizeof(*this)); }
-    error_code load(const std::string &dir) WARN_UNUSED_RESULT;
-    error_code store(const std::string &dir);
+    error_code load(const std::string &dir,
```

Comment:
I don't think it's neccessary to changes these functions of `replica_init_info` or `replica_app_info`. You only need to refactor write_blob_to_file, right?